src/main/reactjs/src/components/elements/CustomModal.tsx

```diff
@@ -0,0 +1,46 @@
+import Box from '@mui/material/Box';
+import Modal from '@mui/material/Modal';
+import { visuallyHidden } from '@mui/utils';
+
+const defaultStyle = {
+  position: 'absolute' as const,
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: '75vw',
+  bgcolor: 'background.paper',
+  border: '2px solid #000',
+  boxShadow: 24,
+  p: 4,
+};
+
+interface CustomModalProps {
+  ariaLabelledBy: string;
+  screenReaderTitle: string;
+  children: JSX.Element;
+  open: boolean;
+  handleCloseModal: () => void;
```

Comment:
handleCloseModal --> onModalClose Olisiko parempi nimi :)

Reply:
onClose löytyy Modal propseista mutta sen signature on:

function(event: object, reason: string) => void
event: The event source of the callback.
reason: Can be: "escapeKeyDown", "backdropClick".

Tässä tapauksessa halutaan siis reagoida tuohon built-in onClose kutsuun, että estetään modaalin sulkeutuminen backdrop clikistä mutta muussa tapauksessa suljetaan sitten modaali kutsumalla onModalClose.

Tilanhallinta on siis parentilla, että voidaan reagoida onnistuneeseen auktorisoinnin lisäämiseen olemassa olevalle kääntäjälle (ao. parentissa):

`  useEffect(() => {
    if (status === APIResponseStatus.Success) {
      handleCloseModal();
    }
  }, [status]);`

Reply:
Joo, mennään tuolla.